### DISCUSSION TITLE
Can not add  "type": "module" in package.json when develop a extension.

### DISCUSSION THREAD

**Discussion by SpeedTest-AI** [microsoft/vscode-discussions#1075] [Extension Development QnA] [answered]
Hi, I am trying to develop a vscode extension which write test code with ai. I need to import the langchain lib. if I do not add {"type":"module"}, it will show error when compile. here is error message  as below.
<img width="1093" alt="image" src="https://github.com/microsoft/vscode-discussions/assets/157690898/fd86686f-9c6c-4ecb-bae7-24de84e5b54c">

if add {"type": "module"} in package.json. compile successfully, but when run command, it show can not found the command.error screenshot as below
<img width="292" alt="image" src="https://github.com/microsoft/vscode-discussions/assets/157690898/9f725e1b-9aee-4926-8923-ccb4b4257582">

here is my code : https://github.com/Test-LLM/ai-test-plugin.git.

Is there way to resolve the problem?  because I want to use the plugin to write auto-test code with AI, so I must import langchain, but the langchain lib file, I can not update it, it seems that I must add {"type":"module"}. but the vscode run in nodejs, it seems that can not add {"type":"module"} in package.

**Comment by isidorn** [comment, marked as answer]
Please read our docs https://code.visualstudio.com/api
